spring-core/src/test/java/org/springframework/util/MultiValueMapRelatedTests.java

```diff
@@ -0,0 +1,214 @@
+		assertThat(objectUnderTest.get("key")).containsExactly("value1", "value2", "value3");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	@Disabled("to be fixed in gh-25140")
+	void addAllWithEmptyList(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.addAll("key", Collections.emptyList());
+		assertThat(objectUnderTest).hasSize(1);
+		assertThat(objectUnderTest.get("key")).isEmpty();
+		assertThat(objectUnderTest.getFirst("key")).isNull();
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void getFirst(MultiValueMap<String, String> objectUnderTest) {
+		List<String> values = new ArrayList<>(2);
+		values.add("value1");
+		values.add("value2");
+		objectUnderTest.put("key", values);
+		assertThat(objectUnderTest.getFirst("key")).isEqualTo("value1");
+		assertThat(objectUnderTest.getFirst("other")).isNull();
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void toSingleValueMap(MultiValueMap<String, String> objectUnderTest) {
+
+		int startingSize = objectUnderTest.size();
+
+		List<String> values = new ArrayList<>(2);
+		values.add("value1");
+		values.add("value2");
+		objectUnderTest.put("key", values);
+		Map<String, String> singleValueMap = objectUnderTest.toSingleValueMap();
+		assertThat(singleValueMap).hasSize(startingSize + 1);
+		assertThat(singleValueMap.get("key")).isEqualTo("value1");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	@Disabled("to be fixed in gh-25140")
+	void toSingleValueMapWithEmptyList(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.put("key", Collections.emptyList());
+		Map<String, String> singleValueMap = objectUnderTest.toSingleValueMap();
+		assertThat(singleValueMap).isEmpty();
+		assertThat(singleValueMap.get("key")).isNull();
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void equalsOnExistingValues(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.clear();
+		objectUnderTest.set("key1", "value1");
+		assertThat(objectUnderTest).isEqualTo(objectUnderTest);
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void equalsOnEmpty(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.clear();
+		objectUnderTest.set("key1", "value1");
+		MultiValueMap<String, String> o1 = new LinkedMultiValueMap<>();
+		o1.set("key1", "value1");
+		assertThat(o1).isEqualTo(objectUnderTest);
+		assertThat(objectUnderTest).isEqualTo(o1);
+		Map<String, List<String>> o2 = new HashMap<>();
+		o2.put("key1", Collections.singletonList("value1"));
+		assertThat(o2).isEqualTo(objectUnderTest);
+		assertThat(objectUnderTest).isEqualTo(o2);
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void canNotChangeAnUnmodifiableMultiValueMap(MultiValueMap<String, String> objectUnderTest) {
+		MultiValueMap<String, String> asUnmodifiableMultiValueMap = CollectionUtils.unmodifiableMultiValueMap(objectUnderTest);
+		Assertions.assertAll(
```

Comment:
It is not permissible to use JUnit Jupiter's `Assertions`. Doing so will actually fail the build.

Thus, please switch to AssertJ assertions and be sure to execute `./gradlew check` to verify that you don't have any Checkstyle violations in this regard.

Note: if you want something similar to `assertAll`, you may wish to use `SoftAssertions` from AssertJ.